portal-kernel/src/com/liferay/portal/kernel/context/InvocationContextProvider.java

```diff
@@ -21,6 +21,8 @@
 
 	public T getCurrent();
 
+	public Class<T> getModelClass();
```

Comment:
What do you think of providing a default implementation like we do in other similar cases using GenericsUtil?
It would look like this:
```
return GenericUtil.getGenericClass(this);
```